### PR TITLE
feat: fetch redirects from udr for migrated repos

### PR DESCRIPTION
**Pull request by LeahMarieBush**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-leah-featudr-redirects-api-hashicorp.vercel.app/) 🔎
- [Asana task](https://app.asana.com/0/1207899860738460/1208961563646273/f) 🎟️

## 🗒️ What

This PR adds support for fetching UDR redirects. It uses the app config to determine if a specific repo is migrated to UDR and if so, pulls the redirects from UDR instead of the product repo. 

## 🤷 Why

To add support for UDR redirects in dev-portal

## 🧪 Testing

Existing functionality:
1. Go to [preview /terraform/cloud-docs/workspaces/creating](https://dev-portal-git-leah-featudr-redirects-api-hashicorp.vercel.app/terraform/cloud-docs/workspaces/creating)
2. Verify the link shown in the search bar is redirected to `/terraform/cloud-docs/workspaces/create`

New functionality: 
1. Run the command below and go to [local /terraform/cloud-docs/workspaces/creating](http://localhost:3000/terraform/cloud-docs/workspaces/creating)
```
HASHI_ENV="unified-docs-sandbox" npm start
```
2. Verify the link shown in the search bar is redirected to `/terraform/cloud-docs/workspaces/create`

### CODE REVIEW

File: build-libs/redirects.js [hunk 1]
```diff
@@ -52,12 +53,28 @@ const HOSTNAME_MAP = {
...
 	 * Note: These constants are declared for clarity in build context intent.
 	 */
 	const isDeveloperBuild = !process.env.IS_CONTENT_PREVIEW
 	const isLocalContentBuild = isDeployPreview(repoName)
+	/**
+	 * Load redirects from the unified docs repo if it's in the list of migrated repos.
+	 * Return early if there are not any redirects found for that specific repo.
+	 */
+	if (
+		config.flags?.unified_docs_migrated_repos?.find((repo) => repo === repoName)
```
Comment (RubenSandwich) [inline build-libs/redirects.js:67]: @LeahMarieBush can you also add a check if we are in the using the unified-docs-sandbox?

process.env.HASHI_ENV === 'unified-docs-sandbox'

File: build-libs/__tests__/redirects.test.ts [hunk 2]
```diff
@@ -239,4 +244,54 @@ describe('filterInvalidRedirects', () => {
 	})
 })
 
+describe('getRedirectsFromContentRepo', () => {
```
Comment (RubenSandwich) [inline build-libs/__tests__/redirects.test.ts:247]: Surprised this function didn't have any testing yet...

**Review by RubenSandwich — CHANGES_REQUESTED** [review]
A small changes noted in this comment:
* https://github.com/hashicorp/dev-portal/pull/2658/files#r1890634025

Also it looks like from redirect test file [that we have to handle a few more .js to .json cases in our UDR migration script](https://github.com/hashicorp/dev-portal/blob/0152c6f97fba7782192d116a3fcd6895f76c8ede/build-libs/__tests__/redirects.test.ts#L50). Namely that we need to add quotes around the following keys:
* key
* value
* has
* type

File: build-libs/redirects.js [hunk 3]
```diff
@@ -52,12 +53,28 @@ const HOSTNAME_MAP = {
...
+	) {
+		const getUDRRedirects = await fetch(
+			`${process.env.UNIFIED_DOCS_API}/api/content/${repoName}/redirects`
+		)
+		if (getUDRRedirects.ok) {
+			const udrRedirects = await getUDRRedirects.json()
+			return udrRedirects
+		}
```
Comment (RubenSandwich) [inline build-libs/redirects.js:75]: @LeahMarieBush If we fail to get a good response for the redirects API we should at least console.error. (If not fail the build, but I'm not sure about failing the build yet. I need to think through that case...)

**Comment by LeahMarieBush** [comment]
> A small changes noted in this comment:
> 
> * https://github.com/hashicorp/dev-portal/pull/2658/files#r1890634025
> 
> Also it looks like from redirect test file [that we have to handle a few more .js to .json cases in our UDR migration script](https://github.com/hashicorp/dev-portal/blob/0152c6f97fba7782192d116a3fcd6895f76c8ede/build-libs/__tests__/redirects.test.ts#L50). Namely that we need to add quotes around the following keys:
> 
> * key
> * value
> * has
> * type

@RubenSandwich I'll get an asana task written to handle these cases in the script 👍 Thanks for pointing them out

**Review by RubenSandwich — APPROVED** [review]
Looks great. 👍🏻

**Pull request merged**